libethash-cl/ethash_cl_miner.cpp

```diff
@@ -495,19 +494,23 @@ void ethash_cl_miner::search(uint8_t const* header, uint64_t target, search_hook
 				pending.pop();
 			}
 
-/*			chrono::high_resolution_clock::duration d = chrono::high_resolution_clock::now() - t;
-			if (d > chrono::milliseconds(_msPerBatch * 10 / 9))
+			// adjust batch size depending on last search time
+			auto d = chrono::duration_cast<chrono::milliseconds>(chrono::high_resolution_clock::now() - t);
+			if (d != chrono::milliseconds(0)) // if duration is zero, we did not get in the actual search so adjust nothing
 			{
```

Comment:
I know this goes against coding standards but as in many other places, this gives a clang warning for potential dangling If, so it asks you to explicitly add the braces.